docs/faq/add_op_in_backend.md

```diff
@@ -0,0 +1,608 @@
+registered as `mxnet.ndarray.quadratic` or `mx.nd.quadratic` for short;
+the other one is for symbolic
+programming registered under module `mxnet.symbol.quadratic`
+or `mx.sym.quadratic` for short.
+
+In order to unit test it in frontend, we need to add the following code
+to the python file `test_operator.py`. Note that while testing the
+forward pass is straightforward using `mx.nd.quadratic`, testing
+the backward involves a bit of more efforts. We create a
+`quadratic` symbol and feed it into the utility function `check_numeric_gradient`.
+The utility function will perform a perturbation on the input
+and calculate the response rate of the output using the
+[finite difference method](https://en.wikipedia.org/wiki/Finite_difference_method).
+Then it will compare the gradient from the backward pass with
+the values from the finite difference method. The test
+will be successful once the comparison satisfies user specified
+relative and absolute thresholds.
+```python
+def test_quadratic_function():
+    def f(x, a, b, c):
+        return a * x**2 + b * x + c
+
+    a = np.random.random_sample()
+    b = np.random.random_sample()
+    c = np.random.random_sample()
+    for ndim in range(1, 6):
+        # check forward
+        shape = rand_shape_nd(ndim, 5)
+        data = rand_ndarray(shape=shape, stype='default')
+        data_np = data.asnumpy()
+        expected = f(data_np, a, b, c)
+        output = mx.nd.quadratic(data, a=a, b=b, c=c)
+        assert_almost_equal(output.asnumpy(), expected)
+
+        # check backward using finite difference
+        data = mx.sym.Variable('data')
+        quad_sym = mx.sym.quadratic(data=data, a=a, b=b, c=c)
+        check_numeric_gradient(quad_sym, [data_np])
+```
+Note that here we used `mx.nd.quadratic` to test the forward function
+and `check_numeric_gradient` to test the backward function. In MXNet,
+two other utility functions are also commonly used: `check_symbolic_forward`
+and `check_symbolic_backward`. By using them in unit tests,
+users need to pass in the operator symbols and expected results
+for comparison. Please also note that
+we highly recommend adding `check_numeric_gradient` test for every operator
+with backward function implemented as it eliminates the possibility
+of passing incorrect expected results into `check_symbolic_backward`.
+
+
+## Summary
+In this tutorial, we practiced implementing the operator `quadratic` in MXNet backend
+and unit testing the implementation in frontend. More specifically, we added parameter
+struct for user-input parameters, walked through shape and type inference workflow,
+implemented forward and backward functions, and registered the operator
+using nnvm. Congratulations! You now know how to add operators.
+We welcome your contributions to MXNet.
+
+**Note**: Source code in the tutorial can be found in
+[quadratic_op-inl.h](https://github.com/reminisce/mxnet/blob/add_op_example_for_tutorial/src/operator/tensor/quadratic_op-inl.h),
```

Comment:
We should remove external dependencies like this. @reminisce can you create a PR for these operators? @thinksanky  please track.

Reply:
I will submit a PR after this is merged and changed the link here.